service/telemetry.go

```diff
@@ -205,12 +214,13 @@ func (tel *telemetryInitializer) initOpenCensus(cfg telemetry.Config, telAttrs m
 
 	tel.views = views
 	if err := view.Register(views...); err != nil {
-		return nil, err
+		return err
 	}
 
 	// Until we can use a generic metrics exporter, default to Prometheus.
 	opts := ocprom.Options{
-		Namespace: "otelcol",
+		Namespace:  "otelcol",
+		Registerer: promRegistry,
```

Comment:
Not sure if it is not better to pass the full `Registry` here. Have not checked the code to see if it has any usage of the Gatherer/Registry. I think it is safer to pass the `Registry: registry`

Reply:
And even maybe use for the moment the Handler from the OC, since I am not 100% if there is no side effect there..

Reply:
The `Gatherer` is just used if we want to use the exporter itself as the handler.

Reply:
:)) famous last words. I am not confident, so still prefer to send the entire Registry to OC and use the Handler form there, since OC is the current state that guarantees me that no new code is used. Look at the PR I created.

Reply:
I did check and couldn't side effects for not using the gatherer. I can double check to make sure, but since we would want to get rid of the oc exporter in the future, we would to create our handler anyway.

Reply:
Sure! I understand your point :)